services/src/main/java/org/keycloak/services/resources/admin/RoleByIdResource.java

```diff
@@ -104,6 +106,11 @@ protected RoleModel getRoleModel(String id) {
     @DELETE
     @NoCache
     public void deleteRole(final @PathParam("role-id") String id) {
+        if (realm.getDefaultRole().getId().equals(id)) {
```

Comment:
The default role should always exist, in case it doesn't the server is in illegal state. But I get your point. 
Not sure if it'll be better to handle `null` possibility here rather than later face NPE or rely on the fact the role has to exist. wdyt?

And if the former the question is what would be correct way to handle it. I lean towards to do nothing: `if (realm.getDefaultRole == null) return;` wdyt?

Reply:
I'd say the right way is to log a warning if the default role does not exist, and then return.

Reply:
@hmlnarik sounds good, thank you

Reply:
Thinking about it a bit deeper, I wonder whether we should not just allow deleting the role in case there is no default role defined. Yes, log a warning, but preventing to delete any role when default role is not set on realm seems too much on the second thought. WDYT?